security/pkg/platform/gcp.go

```diff
@@ -51,9 +53,17 @@ func (j *jwtAccess) RequireTransportSecurity() bool {
 type GcpClientImpl struct {
 	// Root CA cert file to validate the gRPC service in CA.
 	rootCertFile string
-	// Istio CA grpc server
-	caAddr  string
-	fetcher cred.TokenFetcher
+	fetcher      cred.TokenFetcher
+}
+
+func NewGcpClientImpl(rootCertFile, caProvider string, fetcher cred.TokenFetcher) (*GcpClientImpl, error) {
```

Comment:
Doc string is needed.

Reply:
Also, you can check with @quanjielin @myidpt to see if today we can get root cert from GoogleCA automatically.

Can be done as follow-up. leave a todo.

Reply:
I think we can iirc. I will spend more time to address your comments and double check later.

Reply:
And will sync with Quanjie and Oliver